src/main/java/com/github/tomakehurst/wiremock/recording/SnapshotStubMappingPostProcessor.java

```diff
@@ -47,30 +49,46 @@ public SnapshotStubMappingPostProcessor(
   }
 
   public List<StubMapping> process(Iterable<StubMapping> stubMappings) {
-    final Multiset<RequestPattern> requestCounts = HashMultiset.create();
-    final List<StubMapping> processedStubMappings = new ArrayList<>();
-
+    // 1. Run any applicable StubMappingTransformers against the stub mappings.
+    ArrayList<StubMapping> transformedStubMappings = new ArrayList<>();
     for (StubMapping stubMapping : stubMappings) {
-      requestCounts.add(stubMapping.getRequest());
+      transformedStubMappings.add(transformerRunner.apply(stubMapping));
+    }
+
+    // 2. Detect duplicate requests and either discard them or turn them into scenarios.
+    Multiset<RequestPattern> requestCounts = HashMultiset.create();
+    List<StubMapping> processedStubMappings = new ArrayList<>();
+    for (StubMapping transformedStubMapping : transformedStubMappings) {
+      requestCounts.add(transformedStubMapping.getRequest());
 
       // Skip duplicate requests if shouldRecordRepeatsAsScenarios is not enabled
-      if (requestCounts.count(stubMapping.getRequest()) > 1 && !shouldRecordRepeatsAsScenarios) {
+      if (requestCounts.count(transformedStubMapping.getRequest()) > 1
+          && !shouldRecordRepeatsAsScenarios) {
         continue;
       }
 
-      if (bodyExtractMatcher != null
-          && bodyExtractMatcher.match(stubMapping.getResponse()).isExactMatch()) {
-        bodyExtractor.extractInPlace(stubMapping);
-      }
-
-      processedStubMappings.add(stubMapping);
+      processedStubMappings.add(transformedStubMapping);
     }
 
     if (shouldRecordRepeatsAsScenarios) {
       new ScenarioProcessor().putRepeatedRequestsInScenarios(processedStubMappings);
     }
 
-    // Run any stub mapping transformer extensions
-    return processedStubMappings.stream().map(transformerRunner).collect(Collectors.toList());
+    // 3. Extract response bodies to a separate file, if applicable.
+    extractStubMappingBodies(processedStubMappings);
```

Comment:
This is a separate behaviour change and should be part of a separate PR. Also, I think there's a better way of doing this generally, which is to define a size threshold at which stub bodies are extracted as a startup parameter. That way it works in more contexts that just the recorder and allows for more control.